### DISCUSSION TITLE
DataTable Filter using SelectCheckboxMenu

### DISCUSSION THREAD

**Discussion by dhilsbospai** [primefaces/community#486] [PrimeFaces]
We are trying to use a SelectCheckBoxMenu as the filter widget for a column of a DataTable, supplied by a LazyDataModel.  It work great, until we try to supply a default filter (filterBy="#{bb.defaultFilter}").  Without a default filter, the LazyDataModel get a String[] as the value of the filter.  If we set a String[] as the value of a FilterMeta, it used as if it is a Global Filter, and unchecking the boxes in the SelectCheckMenu doesn't change the Filter passed to the LazyDataModel.  I'll put together a minimum example, and add is later, but I'm wondering if anyone has any thoughts right off the bat.

**Comment by FlipWarthog** [comment]
If you're able to set up an example you can @ me to check it. I use SelectCheckBoxMenu filters in a few places, some with defaults, and I've not seen the issue you describe. I don't think I use filterBy on the table itself and we have a lot of custom helper code too, but I should be able to boil it down to the minimum you need.

**Comment by Rapster** [comment]
> It work great, until we try to supply a default filter (filterBy="#{bb.defaultFilter}").

Just to be sure, but default filter is set using `UIColumn#filterValue` not `UIColumn#filterBy`
